doc_src/index.hdr.in

```diff
@@ -17,7 +17,199 @@
+- <a href="#man-page">Manual Pages</a>: Commands usually come with a standardized manual page
+- <a href="#shell-syntax">Command Syntax</a>: Shell commands have a standard syntax
+- <a href="#shell-programming">Commands versus Programs</a>: Commands differ from normal programs
+- <a href="#shebang-line">Shebang Line</a>: How the shell knows the language of a script
+
+\subsection shell-standards Shell Standards
+
+A shell is an interface to the operating system that reads from the commandline of a terminal. A shell's task is to identify and interpret commands. The commands can come from different applications and can be written in different programming languages.
+
+This can only work smoothly if shells adapt to some common standards. For shells there is the POSIX standard, see <a href="https://en.wikipedia.org/wiki/Command-line_interface#Command-line_interpreter">Command-line interpreters</a>. `fish` tries to satisfy the POSIX standard wherever it does not get into the way of its own design principles, see <a href="design.html">`fish` Design</a>.
+
+\subsection man-page Manual Pages
+
+There is a common standard on how to receive help on shell commands: applications provide a manual page to their commands that can be opened with the `man` command:
+
+\fish
+>man COMMAND
+\endfish
+
+This convention helps to make sure help can be found on commands no matter where they originate from. *fish*'s internal commands all come with a manual page.
+
+\subsection shell-syntax Command Syntax
+
+Shells also support some common syntax for executing commands. That way a command can be started in the same way, regardless of the application, where it comes from, and the shell, where it is executed in. 
+
+The pattern below is a basic pattern:
+
+\fish
+COMMAND [OPTIONS] ARGUMENTS
+\endfish
+
+- <b>`COMMAND`</b>: the name of the executeable
+
+- <b>`[OPTIONS]`</b>: options can change the behaviour of a command
+
+- <b>`ARGUMENTS`</b>: input to the command
+
+For external <b>commands</b> the executable is usually stored on file and the file path must be included in the variable `$PATH`, so that the file can be found: see <a href="#variables-special">Special Variables</a>.
+
+<b>Options</b> come in two forms: a short name, that is a hyphen with a single letter; or a long name, consisting of two hyphens with words connected by hyphens. Example: `-h` and `--help` are often used to print a help message. Options are a fixed set, described in the manual pages of the command, see <a href="#man-page">Manual Pages</a>
+
+<b>Arguments</b> are the arbitrary input part of a command: often it is a file or directory name, sometimes it is a string or a list.
+
+Example:
+
+\fish
+>echo -s Hallo World!
+HalloWorld!
+\endfish
+
+-  both `Hallo` and `World!` are arguments to the echo command
+- `-s` is an option that suppresses spaces in the output of the command
+
+\subsection shell-programming Commands versus Programs
```

Comment:
This section confused me. Are you trying to contrast graphical user interface programs with command-line interface programs? I'm not sure how useful a distinction that is to make; plenty of CLI programs are black boxes! Or is this about the difference between shell builtin commands and external programs? Likewise, there's definitely not a clear distinction.

Reply:
yes I probaby got this wrong: I wanted to contrast shell scripting against programming in general. The exit status has a special meaning in shells. And a python program for example can return a variable. A shell script on the other hand can not return a variable. Therefore it is unclear to the novice how to write a shell script as simple as adding two numbers. This is a simple task in most programming languages: you just return a + b, for any give numbers a and b. But in a shell language you cannot return a number. Therefore the only way to do this is to echo the sum and write it to the output stream. This stream can then be picked up by another command. Does this make things clearer to you? I will try to rewrite this section as you suggested.